File: ScorePulse/Models/RehearsalMark.swift
import Foundation

/// Represents a rehearsal mark at a specific bar
struct RehearsalMark: Codable, Identifiable {
    let id: UUID
    let name: String  // e.g. "A", "B", "1", "2", etc.
    let bar: Int
    
    init(id: UUID = UUID(), name: String, bar: Int) {
        self.id = id
        self.name = name
        self.bar = bar
    }
    
    var displayText: String {
        "[\(name)]"
    }
}

File: ScorePulseAndroid/app/src/main/java/com/scorepulse/model/RehearsalMark.kt
package com.scorepulse.model

import kotlinx.serialization.Serializable
import java.util.UUID

/**
 * Represents a rehearsal mark at a specific bar
 */
@Serializable
data class RehearsalMark(
    @Serializable(with = UUIDSerializer::class)
    val id: UUID = UUID.randomUUID(),
    val name: String,
    val bar: Int
) {
    val displayText: String
        get() = "[$name]"
}
